lib/sipper/cli.ex

```diff
@@ -1,17 +1,18 @@
 defmodule Sipper.CLI do
   @config_file System.user_home! <> "/.sipper"
 
-  def main([]) do
+  def main(args) do
+    parse_config_file ++ args
+    |> parse_args
+    |> run
+  end
+
+  defp parse_config_file() do
+    args = []
     if File.exists?(@config_file) do
       args = File.read!(@config_file) |> String.rstrip |> OptionParser.split
-      main(args)
-    else
-      IO.puts :stderr, "Please provide parameters!"
     end
-  end
-
-  def main(args) do
-    args |> parse_args |> run
+    args
   end
```

Comment:
This pattern of modifying a variable inside another scope  is "non-Elixiry" and may start giving warnings/errors at some point: https://github.com/elixir-lang/elixir/issues/3676

The more Elixiry thing is actually less code, too:

``` elixir
  defp config_file_args do
    if File.exists?(@config_file) do
      File.read!(@config_file) |> String.rstrip |> OptionParser.split
    else
      []
    end
  end
```